admin/server/server.go

```diff
@@ -324,3 +333,34 @@ func newURLRegistry(opts *Options) *externalURLs {
 		authLogin:             urlutil.MustJoinURL(opts.ExternalURL, "/auth/login"),
 	}
 }
+
+const publicEmailProvidersListFile = "public_email_providers_list"
+
+//go:embed public_email_providers_list
+var ef embed.FS
+
+// read public email providers list file line by line and load it into a map
+func loadPublicEmailProvidersList() (map[string]bool, error) {
```

Comment:
1. Move the public email logic to `admin/pkg/publicemail`
2. Have a `IsPublic(email)` package-level func that parses the email does the check
3. Use globals and a `sync.Once` to lazily parse the providers into the map
4. Instead of `var ef embed.FS`, the `go:embed` directory supports loading a specific file directly into a string. So you can have something like `var domainsList string`

Reply:
1. Okay
2. Okay
3. I thought it would be better to load file contents into memory at startup rather than at some point during runtime to have predictable memory usage.
4. Wouldn't it cause unnecessary copying into a string first and then to a map ?

Reply:
I wouldn't worry too much about parsing performance here since it's a very minor overhead. Rather, it would just be nice to keep the logic constrained to one package that doesn't have to be separately initialized and passed around as variables (since it's stateless anyway). 

You can use `embed.FS` instead of a direct `string` if you want to reduce that extra bit of memory overhead (the current file looks like ~5kb though?).